hadoop-mini-clusters-hbase/src/main/java/com/github/sakserv/minicluster/impl/HbaseRestLocalCluster.java

```diff
@@ -0,0 +1,363 @@
+        FilterHolder authFilter = pair.getFirst();
+        Class<? extends ServletContainer> containerClass = pair.getSecond();
+        RESTServlet servlet = RESTServlet.getInstance(conf, userProvider);
+
+        // set up the Jersey servlet container for Jetty
+        ServletHolder sh = new ServletHolder(containerClass);
+        sh.setInitParameter("com.sun.jersey.config.property.resourceConfigClass", ResourceConfig.class.getCanonicalName());
+        sh.setInitParameter("com.sun.jersey.config.property.packages", "jetty");
+        ServletHolder shPojoMap = new ServletHolder(containerClass);
+        Map<String, String> shInitMap = sh.getInitParameters();
+        for (Map.Entry<String, String> e : shInitMap.entrySet()) {
+            shPojoMap.setInitParameter(e.getKey(), e.getValue());
+        }
+        shPojoMap.setInitParameter(JSONConfiguration.FEATURE_POJO_MAPPING, "true");
+
+        // set up Jetty and run the embedded server
+
+        server = new Server();
+
+        Connector connector = new SelectChannelConnector();
+        if (conf.getBoolean(RESTServer.REST_SSL_ENABLED, false)) {
+            SslSelectChannelConnector sslConnector = new SslSelectChannelConnector();
+            String keystore = conf.get(RESTServer.REST_SSL_KEYSTORE_STORE);
+            String password = HBaseConfiguration.getPassword(conf, RESTServer.REST_SSL_KEYSTORE_PASSWORD, null);
+            String keyPassword = HBaseConfiguration.getPassword(conf, RESTServer.REST_SSL_KEYSTORE_KEYPASSWORD, password);
+            sslConnector.setKeystore(keystore);
+            sslConnector.setPassword(password);
+            sslConnector.setKeyPassword(keyPassword);
+            connector = sslConnector;
+        }
+        connector.setPort(hbaseRestPort);
+        connector.setHost(hbaseRestHost);
+        connector.setHeaderBufferSize(8192);
+
+
+        server.addConnector(connector);
+
+        QueuedThreadPool threadPool = new QueuedThreadPool(hbaseRestThreadMax);
+        threadPool.setMinThreads(hbaseRestThreadMin);
+        server.setThreadPool(threadPool);
+
+        server.setSendServerVersion(false);
+        server.setSendDateHeader(false);
+        server.setStopAtShutdown(true);
+        // set up context
+        Context context = new Context(server, "/", Context.SESSIONS);
+        context.addServlet(shPojoMap, "/status/cluster");
+        context.addServlet(sh, "/*");
+        if (authFilter != null) {
+            context.addFilter(authFilter, "/*", 1);
+        }
+
+        // Load filters from configuration.
+//        String[] filterClasses = servlet.getConfiguration().getStrings(RESTServer.FILTER_CLASSES, ArrayUtils.EMPTY_STRING_ARRAY);
+//        for (String filter : filterClasses) {
+//            filter = filter.trim();
+//            context.addFilter(Class.forName(filter), "/*", 0);
+//        }
+        RESTServer.addCSRFFilter(context, conf);
```

Comment:
Indeed. If security is disabled, no more problem with RESTServer... ;)
@sakserv : What do you think ?